fondant/dataset.py

```diff
@@ -76,60 +105,124 @@ def load_dataframe(self, spec: FondantComponentSpec) -> dd.DataFrame:
+
+        Raises:
+            ValueError: If a field defined in an output subset is not present in the input
+             dataframe.
+        """
+
+        def verify_subset_columns(subset_name, subset_fields, df):
+            """
+            Verify that all the fields defined in the output subset are present in
+            the input dataframe
+            """
+
+            field_names = list(subset_fields.keys())
+            subset_columns = [f"{subset_name}_{field}" for field in field_names]
             subset_columns.extend(self.mandatory_subset_columns)
 
             for col in subset_columns:
                 if col not in df.columns:
                     raise ValueError(
-                        f"Field {col} defined in output subset {name} but not found in dataframe"
+                        f"Field {col} defined in output subset {subset_name} "
+                        f"but not found in dataframe"
                     )
 
-            # load subset dataframe
+            return subset_columns
+
+        def create_subset_dataframe(subset_name, subset_columns, df):
+            """
+            Create subset dataframe to save with the original field name as the column name
+            """
+            # Create a new dataframe with only the columns needed for the output subset
             subset_df = df[subset_columns]
-            # remove subset prefix from subset columns
+
+            # Remove the subset prefix from the column names
+            prefix_to_replace = f"{subset_name}_"
             subset_df = subset_df.rename(
                 columns={
-                    col: col.split("_")[-1]
+                    col: col[(len(prefix_to_replace)) :]
                     for col in subset_df.columns
                     if col not in self.mandatory_subset_columns
+                    and col.startswith(prefix_to_replace)
                 }
             )
-            # upload to the cloud
-            self._upload_subset(name, subset.fields, subset_df)
+
+            return subset_df
+
+        upload_subsets_tasks = []
+
+        # Loop through each output subset defined in the spec
+        for subset_name, subset in spec.output_subsets.items():
+            # Verify that all the fields defined in the output subset are present in the
+            # input dataframe
+            subset_columns = verify_subset_columns(subset_name, subset.fields, df)
+
+            # Create a new dataframe with only the columns needed for the output subset
+            subset_df = create_subset_dataframe(subset_name, subset_columns, df)
+
+            # Get the remote path where the output subset should be uploaded
+            remote_path = self.manifest.subsets[subset_name].location
+
+            # Create the expected schema for the output subset
+            expected_schema = {
+                field.name: field.type.name for field in subset.fields.values()
```

Comment:
Don't we need to convert these types to `PyArrow` types?

Reply:
not necessarily, it works fine with general parquet types (similar to how it's currently defined for the `index_schema`)

Reply:
Ok, not sure if the `field.type.name` completely align with `parquet` types though.

Reply:
They currently don't align completely with the `Arrow` data types either. If those are the ones we need, we can update our types to match the `Arrow` types completely. But we can do this in a separate PR. I'll add a ticket.

Reply:
At first glance it seems like they align no? I'm comparing both of these: https://github.com/ml6team/fondant/blob/main/fondant/schema.py
https://arrow.apache.org/docs/python/api/datatypes.html#api-types

Reply:
At least `categorical` is not aligned, and we are missing some.